src/Orleans/Runtime/GrainReference.cs

```diff
@@ -531,14 +551,15 @@ protected internal static object DeserializeGrainReference(Type t, IDeserializat
             }
             // store as null, serialize as empty.
             var genericArg = reader.ReadString();
-            if (String.IsNullOrEmpty(genericArg))
+            if (string.IsNullOrEmpty(genericArg))
                 genericArg = null;
-
+            
             if (expectObserverId)
             {
-                return NewObserverGrainReference(id, observerId);
+                return NewObserverGrainReference(id, observerId, Runtime.RuntimeClient.Current);
```

Comment:
Currently, the global `RuntimeClient.Current` is used. In a later PR, that will be changed to retrieve the `IRuntimeClient` from the context.